src/Language/DifferentialDatalog/Rule.hs

```diff
@@ -73,7 +75,12 @@ rulePPPrefix rl len = commaSep $ map pp $ take len $ ruleRHS $ ruleStripTypeAnno
 
 -- | Pretty-print rule without type annotations.
 rulePPStripped :: Rule -> Doc
-rulePPStripped rl = pp $ ruleStripTypeAnnotations rl
+rulePPStripped rule = pp (ruleStripTypeAnnotations rule) <+> char '@' <+> file <> char ':' <> line <> char ':' <> column
```

Comment:
What's wrong with `pp $ rulePos rule`?

Reply:
I don't know, I don't think it has the same output, does it?

Reply:
My bad, I was thinking of this function in `Pos.hs`.  It prints the entire range, not just its start.

```
spos :: (WithPos a) => a -> String
spos x = let (s,e) = pos x
         in sourceName s ++ ":" ++ (show $ sourceLine s) ++ "." ++ (show $ sourceColumn s) ++ "-"
                                ++ (show $ sourceLine e) ++ "." ++ (show $ sourceColumn e)
```